framework/db/src/com/cloud/utils/db/Transaction.java

```diff
@@ -59,4 +59,30 @@ public T doInTransaction(TransactionStatus status) throws RuntimeException {
         });
     }
 
+    @SuppressWarnings("deprecation")
+    public static <T, E extends Throwable> T execute(final short databaseId, TransactionCallbackWithException<T, E> callback) throws E {
+        String name = "tx-" + counter.incrementAndGet();
+        TransactionLegacy currentTxn = TransactionLegacy.currentTxn(false);
+        short outer_txn_databaseId = (currentTxn != null ? currentTxn.getDatabaseId() : databaseId);
+        TransactionLegacy txn = TransactionLegacy.open(name, databaseId, true);
+        try {
```

Comment:
Lines 67 and 68 can be combined into a try with resources as follows:

```
        try (final TransactionLegacy txn = TransactionLegacy.open(name, databaseId, false)) {
```

This construct also removes the need for the ```txn.close()``` on line 74.